OKTA-508890 Update the android browser redirect guide to okta-mobile-kotlin. (#3464)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md

> 1. Update the default layout file for the app (found in `app/src/main/res/layout/`) to add a sign-in button, sign-out button, and a text view:

Reviewer: A slight order change... and what's the file?? There's just a path:
```suggestion
1. Add sign-in and sign-out buttons and a text view to the app's default layout file by adding the following code to `app/src/main/res/layout/`:
```

        >

            <TextView
                android:id="@+id/status_text_view"
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                android:layout_marginTop="16dp"
                />

            <Button
                android:id="@+id/login_button"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_marginTop="16dp"
                android:text="@string/login_button"
                />

            <Button
                android:id="@+id/logout_of_browser_button"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_marginTop="16dp"
                android:backgroundTint="@android:color/holo_red_light"
                android:text="@string/logout_of_browser_button"
                />

            <ProgressBar
                android:layout_gravity="center"
                android:id="@+id/progress_bar"
                android:layout_width="50dp"
                android:layout_height="50dp"
                />
    </LinearLayout>
    ```

> 1. Fill in strings for the `text` values referenced above, in `app/src/main/res/values/strings.xml`:

Reviewer: ```suggestion
1. Add the following code to `app/src/main/res/values/strings.xml` to add the strings for the `text` values of the buttons:
```

        ) : BrowserState() {
            companion object {
                /**
                 * Creates the [LoggedIn] state using the [CredentialBootstrap.defaultCredential]s ID Token name claim.
                 */
                suspend fun create(errorMessage: String? = null): BrowserState {
                    val credential = CredentialBootstrap.defaultCredential()
                    val name = credential.idToken()?.name ?: ""
                    return LoggedIn(name, errorMessage)
                }
            }
        }

        companion object {
            /**
             * Creates the [BrowserState] given the [CredentialBootstrap.defaultCredential]s presence of a token.
             *
             * @return Either [LoggedIn] or [LoggedOut].
             */
            suspend fun currentCredentialState(errorMessage: String? = null): BrowserState {
                val credential = CredentialBootstrap.defaultCredential()
                return if (credential.token == null) {
                    LoggedOut(errorMessage)
                } else {
                    LoggedIn.create(errorMessage)
                }
            }
        }
    }
    ```

> 1. Add the following content to your activity (for example `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.kt`) to handle button clicks and browser state events:

Reviewer: Be prescriptive for filenames and classnames:
```suggestion
1. Add the following code Thant handles button clicks and browser state events to `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.kt`:
```

                        } else {
                            statusTextView.text = state.errorMessage
                        }
                    }
                    is BrowserState.LoggedOut -> {
                        progressBar.visibility = View.GONE
                        loginButton.visibility = View.VISIBLE
                        logoutOfBrowserButton.visibility= View.GONE
                        statusTextView.visibility = View.VISIBLE
                        if (state.errorMessage == null) {
                            statusTextView.text = getString(R.string.have_account)
                        } else {
                            statusTextView.text = state.errorMessage
                        }
                    }
                    BrowserState.Loading -> {
                        progressBar.visibility = View.VISIBLE
                        loginButton.visibility = View.GONE
                        logoutOfBrowserButton.visibility= View.GONE
                        statusTextView.visibility = View.GONE
                    }
                }
            }
        }
    }
    ```

#### Running the sample

> To run the sample, see [the Android Studio documentation](https://developer.android.com/studio/run/device).

Reviewer: Is this needed? Same reason I asked before, in the iOS world developers are assumed to know Xcode and how to run on a device versus Simulator
